corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -452,28 +452,17 @@ def clean(self):
         report won't be able to use one of the existing ones.
         """
         cleaned_data = super(DataSourceForm, self).clean()
-        source_type = cleaned_data.get('source_type')
-        report_source = cleaned_data.get('source')
-        app_id = cleaned_data.get('application')
-
-        if report_source and source_type and app_id:
-
-            app = Application.get(app_id)
-            ds_builder = DataSourceBuilder(self.domain, app, source_type, report_source)
-
-            existing_sources = DataSourceConfiguration.by_domain(self.domain)
-            active_source = filter(lambda config: not config.is_deactivated, existing_sources)
-            if len(active_source) >= 5:
-                match = ds_builder.get_existing_match()
-                if not match or match.is_deactivated:
-                    raise forms.ValidationError(_(
-                        "Too many data sources!\n"
-                        "Creating this report would cause you to go over the maximum "
-                        "number of data sources allowed in this domain. The current "
-                        "limit is 5. "
-                        "To continue, delete all of the reports using a particular "
-                        "data source (or the data source itself) and try again. "
-                    ))
+
+        existing_reports = ReportConfiguration.by_domain(self.domain)
+        builder_reports = filter(lambda report: report.report_meta.created_by_builder, existing_reports)
+        if len(builder_reports) >= 5:
```

Comment:
@czue I added a feature flag in https://github.com/dimagi/commcare-hq/pull/10664 so that current domains that needed to be over the limit could (prompted by a training that needed to occur). Once we have a firmer idea how we want to implement that feature (particularly from a ui/pricing perspective), I can convert it to being a number saved on the domain.